js-api-spec/value/calculation.test.ts

```diff
@@ -0,0 +1,424 @@
+// Copyright 2023 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import {
+  Value,
+  SassCalculation,
+  SassNumber,
+  SassString,
+  CalculationOperation,
+  CalculationOperator,
+  CalculationInterpolation,
+  compileString,
+  compileStringAsync,
+} from 'sass';
+import {List} from 'immutable';
+
+import '../utils';
+
+const validCalculationValues = [
+  new SassNumber(1),
+  new SassString('1', {quotes: false}),
+  SassCalculation.calc(new SassNumber(1)),
+  new CalculationOperation('+', new SassNumber(1), new SassNumber(1)),
+  new CalculationInterpolation(''),
+];
+const invalidCalculationValues = [new SassString('1', {quotes: true})];
+
+describe('SassCalculation', () => {
+  describe('construction', () => {
+    let calculation: SassCalculation;
+    beforeEach(() => {
+      calculation = SassCalculation.calc(new SassNumber(1));
+    });
+
+    it('is a value', () => {
+      expect(calculation).toBeInstanceOf(Value);
+    });
+
+    it('is a calculation', () => {
+      expect(calculation).toBeInstanceOf(SassCalculation);
+      expect(calculation.assertCalculation()).toBe(calculation);
+    });
+
+    it("isn't any other type", () => {
+      expect(() => calculation.assertBoolean()).toThrow();
+      expect(() => calculation.assertColor()).toThrow();
+      expect(() => calculation.assertFunction()).toThrow();
+      expect(() => calculation.assertMap()).toThrow();
+      expect(calculation.tryMap()).toBe(null);
+      expect(() => calculation.assertNumber()).toThrow();
+      expect(() => calculation.assertString()).toThrow();
+    });
```

Comment:
Can you add `assertCalculation()` to the "isn't any other type" tests for all the other types?

Reply:
I think I did on 17c55ef9c064ccc29ebce1d6e33837b5699edd58, unless I missed something?

Reply:
Oh, but they were missing from the dart test suite. I added them in https://github.com/oddbird/dart-sass/commit/6b3b5b80708ce47fdc5af4b4611aa4239c497ad4 and https://github.com/sass/dart-sass/pull/1988/commits/52ab266749a3e6c3a18301afb3c9549592eace94